src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1319,6 +1321,175 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
   return r;
 }
 
+int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &current,
+                                 const boost::optional<Snapshot>& prev) {
```

Comment:
Can just be `boost::optional<Snapshot> prev`, correct?

Reply:
Does it need to be `optional` now? Can it be a `const Snapshot&`?

Reply:
`prev` is optional (case where the mirror daemon is synchronizing the very first snapshot), isn't it?

Reply:
updated